File: app/src/main/java/com/odogwudev/first/MainActivity.java
package com.odogwudev.first;

import androidx.appcompat.app.AppCompatActivity;

import android.util.Pair;

import android.app.ActivityOptions;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.view.WindowManager;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private static int SPLASH_SCREEN = 5000;

    Animation topAnim, bottomAnim;
    ImageView image;
    TextView title, slogan;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getWindow().setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN
        );

        // Animations
        topAnim = AnimationUtils.loadAnimation(this, R.anim.top_animation);
        bottomAnim = AnimationUtils.loadAnimation(this, R.anim.bottom_animation);

        image = findViewById(R.id.ivCar);
        title = findViewById(R.id.tvTitle);
        slogan = findViewById(R.id.tvSlogan);

        image.setAnimation(topAnim);
        title.setAnimation(bottomAnim);
        slogan.setAnimation(bottomAnim);

        new Handler().postDelayed(() -> {

            Intent intent = new Intent(MainActivity.this, LoginActivity.class);

            Pair[] pairs = new Pair[2];
            pairs[0] = new Pair<View, String>(image, "logo_image");
            pairs[1] = new Pair<View, String>(title, "logo_text");

            ActivityOptions options = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, pairs);
            startActivity(intent, options.toBundle());

        }, SPLASH_SCREEN);

    }
}

File: app/src/main/java/com/odogwudev/first/MainActivity.kt
package com.odogwudev.first

import androidx.appcompat.app.AppCompatActivity;
import android.util.Pair;

import android.app.ActivityOptions;
import android.content.Intent;
import android.os.Build
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper
import android.view.View;
import android.view.WindowInsets
import android.view.WindowManager;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        @Suppress("DEPRECATION")
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.insetsController?.hide(WindowInsets.Type.statusBars())
        } else {
            window.setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN
            )


            val title: TextView = findViewById(R.id.tvTitle)
            val slogan: TextView = findViewById(R.id.tvSlogan)
            val image: ImageView = findViewById(R.id.ivCar)

            val topAnim = AnimationUtils.loadAnimation(this@MainActivity, R.anim.top_animation)
            val bottomAnim =
                AnimationUtils.loadAnimation(this@MainActivity, R.anim.bottom_animation)


            val pair = listOf<Pair<View, String>>(
                Pair<View, String>(image, "logo_image"),
                Pair<View, String>(title, "logo_text"),
            )
            val options: ActivityOptions =
                ActivityOptions.makeSceneTransitionAnimation(this, pair[0], pair[1])
            Intent(this, MainActivity::class.java).also {
                startActivity(it, options.toBundle())
            }


            Handler(Looper.getMainLooper()).postDelayed({
                startActivity(Intent(this@MainActivity, LoginActivity::class.java))
                overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
                finish()
            }, 5000)
        }


    }
}